src/parser/binary-operation-parser.ts

```diff
@@ -1,4 +1,4 @@
-import { FilterNode } from '../operation-node/filter-node.js'
+import { BinaryOperationNode } from '../operation-node/binary-operation-node.js'
```

Comment:
Can we really generalize this whole file from filter-parser to binary-operation-parser? I don't think so. I think this should still remain `filter-parser` until we actually implement something other than filters using the new nodes.